core/Enclave/enclave_hsm.cpp

```diff
@@ -995,11 +1255,57 @@ sgx_status_t enclave_generate_hmac(ehsm_keyblob_t *cmk, uint32_t cmk_size,
     ret = sgx_hmac_sha256_msg(payload->data, payload->datalen, rawApiKey->data, rawApiKey->datalen, hmac->data, hmac->datalen);
 
 out:
-    // clear sensitive info
-    if (rawApiKey) {
-        memset_s(rawApiKey, APPEND_SIZE_TO_DATA_T(rawApiKey->datalen), 0, APPEND_SIZE_TO_DATA_T(rawApiKey->datalen));
-    }
-    // free allocations
+    SAFE_MEMSET(rawApiKey, APPEND_SIZE_TO_DATA_T(rawApiKey->datalen), 0, APPEND_SIZE_TO_DATA_T(rawApiKey->datalen));
     SAFE_FREE(rawApiKey);
     return ret;
+}
+
+sgx_status_t enclave_generate_token_hmac(ehsm_keyblob_t *cmk, uint32_t cmk_size,
```

Comment:
cmk mean customer master key. 

Here your cmk refers to the key to generate HMAC signature, right?

Reply:
fix